src/ethereum/osaka/vm/instructions/bitwise.py

```diff
@@ -238,3 +238,31 @@ def bitwise_sar(evm: Evm) -> None:
 
     # PROGRAM COUNTER
     evm.pc += Uint(1)
+
+
+def count_leading_zeros(evm: Evm) -> None:
+    """
+    Count the number of leading zero bits in a 256-bit word.
+
+    Pops one value from the stack and pushes the number of leading zero bits.
+    If the input is zero, pushes 256.
+
+    Parameters
+    ----------
+    evm :
+        The current EVM frame.
+    """
+    # STACK
+    x = pop(evm.stack)
+
+    # GAS
+    charge_gas(evm, GAS_VERY_LOW)
```

Comment:
Small nit: we want to charge gas before doing any operations. So the gas should be charged before the `pop`. If there is not enough gas available, then we thus also do not perform the `pop` logic where there are insufficient funds for :smile: :+1: